controllers/druid/handler.go

```diff
@@ -1302,30 +1302,58 @@ func makePodTemplate(nodeSpec *v1alpha1.DruidNodeSpec, m *v1alpha1.Druid, ls map
 
 // makePodSpec shall create podSpec common to both deployment and statefulset.
 func makePodSpec(nodeSpec *v1alpha1.DruidNodeSpec, m *v1alpha1.Druid, nodeSpecUniqueStr, configMapSHA string) v1.PodSpec {
-	spec := v1.PodSpec{
-		NodeSelector:              firstNonNilValue(m.Spec.NodeSelector, nodeSpec.NodeSelector).(map[string]string),
-		TopologySpreadConstraints: getTopologySpreadConstraints(nodeSpec),
-		Tolerations:               getTolerations(nodeSpec, m),
-		Affinity:                  getAffinity(nodeSpec, m),
-		ImagePullSecrets:          firstNonNilValue(nodeSpec.ImagePullSecrets, m.Spec.ImagePullSecrets).([]v1.LocalObjectReference),
-		Containers: []v1.Container{
-			{
-				Image:           firstNonEmptyStr(nodeSpec.Image, m.Spec.Image),
-				Name:            fmt.Sprintf("%s", nodeSpecUniqueStr),
-				Command:         []string{firstNonEmptyStr(m.Spec.StartScript, "bin/run-druid.sh"), nodeSpec.NodeType},
-				ImagePullPolicy: v1.PullPolicy(firstNonEmptyStr(string(nodeSpec.ImagePullPolicy), string(m.Spec.ImagePullPolicy))),
-				Ports:           nodeSpec.Ports,
-				Resources:       nodeSpec.Resources,
-				Env:             getEnv(nodeSpec, m, configMapSHA),
-				EnvFrom:         getEnvFrom(nodeSpec, m),
-				VolumeMounts:    getVolumeMounts(nodeSpec, m),
-				LivenessProbe:   getLivenessProbe(nodeSpec, m),
-				ReadinessProbe:  getReadinessProbe(nodeSpec, m),
-				StartupProbe:    getStartUpProbe(nodeSpec, m),
-				Lifecycle:       nodeSpec.Lifecycle,
-				SecurityContext: firstNonNilValue(nodeSpec.ContainerSecurityContext, m.Spec.ContainerSecurityContext).(*v1.SecurityContext),
-			},
+
+	var containers []v1.Container
+	containers = append(containers,
+		v1.Container{
+			Image:           firstNonEmptyStr(nodeSpec.Image, m.Spec.Image),
+			Name:            fmt.Sprintf("%s", nodeSpecUniqueStr),
+			Command:         []string{firstNonEmptyStr(m.Spec.StartScript, "bin/run-druid.sh"), nodeSpec.NodeType},
+			ImagePullPolicy: v1.PullPolicy(firstNonEmptyStr(string(nodeSpec.ImagePullPolicy), string(m.Spec.ImagePullPolicy))),
+			Ports:           nodeSpec.Ports,
+			Resources:       nodeSpec.Resources,
+			Env:             getEnv(nodeSpec, m, configMapSHA),
+			EnvFrom:         getEnvFrom(nodeSpec, m),
+			VolumeMounts:    getVolumeMounts(nodeSpec, m),
+			LivenessProbe:   getLivenessProbe(nodeSpec, m),
+			ReadinessProbe:  getReadinessProbe(nodeSpec, m),
+			StartupProbe:    getStartUpProbe(nodeSpec, m),
+			Lifecycle:       nodeSpec.Lifecycle,
+			SecurityContext: firstNonNilValue(nodeSpec.ContainerSecurityContext, m.Spec.ContainerSecurityContext).(*v1.SecurityContext),
 		},
+	)
+
+	if m.Spec.SidecarContainer != nil {
+
+		for _, containerList := range m.Spec.SidecarContainer {
+			configs := containerList.Configs
+			args := containerList.Args
+			resource := containerList.Resources
+
+			containers = append(containers,
+				v1.Container{
+					Image:           configs["imageName"],
```

Comment:
can we add a constants blob with respect to all the strings, something like
```
const (
  ImageName = "imageName"
  
)
```
just to make it cleaner. you can add it in interface.go.